iree/turbine/kernel/wave/wave.py

```diff
@@ -136,6 +148,44 @@ def _warn_iree_is_too_old():
         )
 
 
+def build_wave_runtime():
+    wave_runtime_lib = "wave_runtime*.so"
+    lib_path = glob.glob(str(WAVE_RUNTIME_DIR / wave_runtime_lib))
+    # Don't build if lib already exists.
+    if lib_path and Path(lib_path[0]).is_file():
+        return
+
+    # Run cmake command.
+    runtime_path = Path(__file__).resolve()
+    runtime_path = runtime_path.parent / "runtime"
+    runtime_build_path = runtime_path / "build"
+    args = [
+        "cmake",
+        f"-DPython_EXECUTABLE={sys.executable}",
+        "-S",
+        str(runtime_path),
+        "-B",
+        str(runtime_build_path),
+    ]
+    try:
+        subprocess.run(args, check=True)
```

Comment:
Python should not be running CMake commands on the system at runtime. This needs a system architecture discussion.

Reply:
Yeah, it's a plan of record to move this to setup.py. IIRC this was semi inspired by triton, who's runtime is compiled by calling clang from python haha

Reply:
Yes, this was a stepping stone to prove this concept. Now that we have proved its worth, the plan is to move it to setup.py.

Reply:
iree-turbine is currently a pure Python package, portable across all operating systems and python versions. Ignoring all details about what the new native code is doing (and those are certainly nontrivial), that immediately introduces additional complexity for building, testing, and packaging in this repository.

There are multiple different ways to take the various packages involved here, hence this needs a system architecture discussion. That should ideally have happened before merging the code to `main`, and I'm personally not comfortable with iterating on the `main` branch without some strong signaling in documentation and in code that the new components are optional and experimental.